### PR TITLE
dissect-image: prefer PARTN= uevent property over "partition" sysfs attr

### PR DESCRIPTION
**Pull request by poettering**
The kernel will send us a PARTN= uevent proprty with partition add
events, let's use it instead of going for the "partition" sysfs attr.
It's less racy that way and there are reports the sysfs attr shows up
after the device, which makes it evern worse.

### CODE REVIEW

File: src/shared/dissect-image.c [hunk 1]
```diff
@@ -174,9 +178,16 @@ static int device_is_partition(sd_device *d, sd_device *expected_parent, blkid_p
         if (!path_equal(parent_syspath, expected_parent_syspath))
                 return false; /* Has a different parent than what we need, not interesting to us */
 
-        r = sd_device_get_sysattr_value(d, "partition", &v);
+        /* On kernel uevents the partition number we may find in the PARTN= field. Let's use that preferably,
```
Comment (bluca) [inline src/shared/dissect-image.c:181]: "we may find the partition number in the PARTN= field"

File: src/shared/dissect-image.c [hunk 2]
```diff
@@ -174,9 +178,16 @@ static int device_is_partition(sd_device *d, sd_device *expected_parent, blkid_p
...
+         * since it's cheaper and more importantly: the sysfs attribute "partition" appear to become
```
Comment (bluca) [inline src/shared/dissect-image.c:182]: /appear/appears/

**Review by bluca — APPROVED** [review]
just some spelling nits, LGTM

**Comment by poettering** [comment]
I accidentally merged this before addressing @bluca's points. Done so now in #19400.

**Pull request merged**